File: eCommerce/UI/ecommerce_mobile/lib/model/restaurant.dart
class Restaurant {
  final int id;
  final int ownerId;
  final String ownerName;
  final String name;
  final String? description;
  final String address;
  final int cityId;
  final String cityName;
  final double? latitude;
  final double? longitude;
  final String? phoneNumber;
  final String? email;
  final int cuisineTypeId;
  final String cuisineTypeName;
  final double? averageRating;
  final int totalReviews;
  final bool hasParking;
  final bool hasTerrace;
  final bool isKidFriendly;
  final String openTime; // Format: "HH:mm:ss"
  final String closeTime; // Format: "HH:mm:ss"
  final DateTime createdAt;
  final bool isActive;

  Restaurant({
    required this.id,
    required this.ownerId,
    required this.ownerName,
    required this.name,
    this.description,
    required this.address,
    required this.cityId,
    required this.cityName,
    this.latitude,
    this.longitude,
    this.phoneNumber,
    this.email,
    required this.cuisineTypeId,
    required this.cuisineTypeName,
    this.averageRating,
    required this.totalReviews,
    required this.hasParking,
    required this.hasTerrace,
    required this.isKidFriendly,
    required this.openTime,
    required this.closeTime,
    required this.createdAt,
    required this.isActive,
  });

  factory Restaurant.fromJson(Map<String, dynamic> json) {
    return Restaurant(
      id: json['id'] ?? 0,
      ownerId: json['ownerId'] ?? 0,
      ownerName: json['ownerName'] ?? '',
      name: json['name'] ?? '',
      description: json['description'],
      address: json['address'] ?? '',
      cityId: json['cityId'] ?? 0,
      cityName: json['cityName'] ?? '',
      latitude: json['latitude'] != null ? (json['latitude'] as num).toDouble() : null,
      longitude: json['longitude'] != null ? (json['longitude'] as num).toDouble() : null,
      phoneNumber: json['phoneNumber'],
      email: json['email'],
      cuisineTypeId: json['cuisineTypeId'] ?? 0,
      cuisineTypeName: json['cuisineTypeName'] ?? '',
      averageRating: json['averageRating'] != null ? (json['averageRating'] as num).toDouble() : null,
      totalReviews: json['totalReviews'] ?? 0,
      hasParking: json['hasParking'] ?? false,
      hasTerrace: json['hasTerrace'] ?? false,
      isKidFriendly: json['isKidFriendly'] ?? false,
      openTime: json['openTime'] ?? '09:00:00',
      closeTime: json['closeTime'] ?? '22:00:00',
      createdAt: json['createdAt'] != null ? DateTime.parse(json['createdAt']) : DateTime.now(),
      isActive: json['isActive'] ?? true,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'ownerId': ownerId,
      'ownerName': ownerName,
      'name': name,
      'description': description,
      'address': address,
      'cityId': cityId,
      'cityName': cityName,
      'latitude': latitude,
      'longitude': longitude,
      'phoneNumber': phoneNumber,
      'email': email,
      'cuisineTypeId': cuisineTypeId,
      'cuisineTypeName': cuisineTypeName,
      'averageRating': averageRating,
      'totalReviews': totalReviews,
      'hasParking': hasParking,
      'hasTerrace': hasTerrace,
      'isKidFriendly': isKidFriendly,
      'openTime': openTime,
      'closeTime': closeTime,
      'createdAt': createdAt.toIso8601String(),
      'isActive': isActive,
    };
  }
}

File: eCommerce/eCommerce.Services/Database/Restaurant.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace eCommerce.Services.Database
{
    public class Restaurant
    {
        [Key]
        public int Id { get; set; }
        
        public int OwnerId { get; set; }
        
        [ForeignKey("OwnerId")]
        public User Owner { get; set; } = null!;
        
        [Required]
        [MaxLength(100)]
        public string Name { get; set; } = string.Empty;
        
        [MaxLength(1000)]
        public string? Description { get; set; }
        
        [Required]
        [MaxLength(200)]
        public string Address { get; set; } = string.Empty;
        
        public int CityId { get; set; }
        
        [ForeignKey("CityId")]
        public City City { get; set; } = null!;
        
        [Column(TypeName = "decimal(9,6)")]
        public decimal? Latitude { get; set; }
        
        [Column(TypeName = "decimal(9,6)")]
        public decimal? Longitude { get; set; }
        
        [Phone]
        [MaxLength(20)]
        public string? PhoneNumber { get; set; }
        
        [EmailAddress]
        [MaxLength(100)]
        public string? Email { get; set; }
        
        public int CuisineTypeId { get; set; }
        
        [ForeignKey("CuisineTypeId")]
        public CuisineType CuisineType { get; set; } = null!;
        
        [Column(TypeName = "decimal(3,2)")]
        public decimal? AverageRating { get; set; }
        
        public int TotalReviews { get; set; } = 0;
        
        public bool HasParking { get; set; } = false;
        
        public bool HasTerrace { get; set; } = false;
        
        public bool IsKidFriendly { get; set; } = false;
        
        public TimeSpan OpenTime { get; set; }
        
        public TimeSpan CloseTime { get; set; }
        
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        
        public bool IsActive { get; set; } = true;
        
        // Navigation properties
        public ICollection<Table> Tables { get; set; } = new List<Table>();
        public ICollection<Reservation> Reservations { get; set; } = new List<Reservation>();
        public ICollection<RestaurantGallery> Gallery { get; set; } = new List<RestaurantGallery>();
        public ICollection<MenuItem> MenuItems { get; set; } = new List<MenuItem>();
        public ICollection<SpecialOffer> SpecialOffers { get; set; } = new List<SpecialOffer>();
        public ICollection<Review> Reviews { get; set; } = new List<Review>();
        public ICollection<RestaurantStatistic> Statistics { get; set; } = new List<RestaurantStatistic>();
        public ICollection<ChatConversation> ChatConversations { get; set; } = new List<ChatConversation>();
        public ICollection<Reward> Rewards { get; set; } = new List<Reward>();
    }
}

